src/linker/configuration.py

```diff
@@ -32,24 +32,28 @@ def __init__(
         self.computing_environment = self.environment["computing_environment"]
         self.container_engine = self.environment.get("container_engine", None)
         self.steps = self._get_steps()
+        self.implementation_metadata = self._load_implementation_metadata()
 
-    def get_step_directory(self, step_name: str) -> Path:
+    def get_implementation_directory(self, step_name: str) -> Path:
         # TODO: move this into proper config validator
         implementation = self.pipeline["steps"][step_name]["implementation"]
         implementation_dir = (
-            Path(os.path.realpath(__file__)).parent.parent.parent
-            / "steps"
-            / step_name
-            / "implementations"
+            Path(os.path.realpath(__file__)).parent / "steps" / step_name / "implementations"
         )
-        implementation_names = next(os.walk(implementation_dir))[1]
+        implementation_names = [
+            str(d.name) for d in implementation_dir.iterdir() if d.is_dir()
+        ]
```

Comment:
As discussed earlier, I think this will move out of this file after the refactors in your next ticket.